### DISCUSSION TITLE
TypeError: Invalid URL when using next-auth in  CI pipeline Vercel preview deployment "

### DISCUSSION THREAD

**Discussion by rukawa444** [nextauthjs/next-auth#7801] [Help]
### Reproduction URL

https://github.com/cathleys/cath.to/actions/runs/5264917271/jobs/9516838827?pr=23

### Describe the issue

Hey guys, I am new to everything about Next js ecosystem and stuff as I'm building my own project. I can't seem to figure out the error myself.

I am having issues with my CI setup containing a Vercel preview deployment which throws an issue regarding the parseURL.

**This is the error details in the CI:**

BUILD PROJECT ARTIFACT
//...
info  - Linting and checking validity of types...
info  - Creating an optimized production build...
info  - Compiled successfully
info  - Collecting page data...

**TypeError [ERR_INVALID_URL]: Invalid URL**
    at new NodeError (node:internal/errors:399:5)
    at new URL (node:internal/url:560:13)
    at parseUrl (/home/runner/work/cath.to/cath.to/node_modules/next-auth/utils/parse-url.js:17:16)
    at Object.<anonymous> (/home/runner/work/cath.to/cath.to/node_modules/next-auth/react/index.js:72:[34](https://github.com/cathleys/cath.to/actions/runs/5264917271/jobs/9516838827?pr=23#step:5:35))
    at Module._compile (node:internal/modules/cjs/loader:1254:14)
    at Module._extensions..js (node:internal/modules/cjs/loader:1308:10)
    at Module.load (node:internal/modules/cjs/loader:1117:32)
    at Module._load (node:internal/modules/cjs/loader:958:12)
    at Module.require (node:internal/modules/cjs/loader:11[41](https://github.com/cathleys/cath.to/actions/runs/5264917271/jobs/9516838827?pr=23#step:5:42):19)
    at require (node:internal/modules/cjs/helpers:110:18) {
  input: '',
  code: 'ERR_INVALID_URL'
}

> Build error occurred
Error: Failed to collect page data for /create-post
    at /home/runner/work/cath.to/cath.to/node_modules/next/dist/build/utils.js:916:15
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5) {
  type: 'Error'
}
Error: Command "npm run build" exited with 1
Error: Process completed with exit code 1.`

My **package.json** file looks like this:

 "scripts": {
    "dev": "next dev",
    "build": "next build",
    "start": "next start",
    "lint": "eslint .",
    "prettier": "prettier --check .",
    "storybook": "storybook dev -p 6006",
    "build-storybook": "storybook build",
    "prepare": "husky install"
  },
  "lint-staged": {
    "*.{js,jsx,ts,tsx}": "eslint --cache --fix",
    "*.{ts,tsx}": "bash -c 'tsc --noEmit'",
    "*.{js,jsx,ts,tsx,css,md}": "prettier --write"
  },
  "dependencies": {
    "@emotion/react": "^11.11.0",
    "@emotion/styled": "^11.11.0",
    "@fontsource-variable/inter": "^5.0.2",
    "@mui/material": "^5.13.3",
    "@types/node": "20.2.5",
    "@types/react": "18.2.7",
    "@types/react-dom": "18.2.4",
    "axios": "^1.4.0",
    "bcryptjs": "^2.4.3",
    "dotenv": "^16.1.4",
    "eslint": "8.41.0",
    "eslint-config-next": "^12.0.0",
    "mongoose": "^7.2.2",
    **"next": "^12.0.0",**
    **"next-auth": "^4.22.1",**
    "react": "18.2.0",
    "react-dom": "18.2.0",
    "react-icons": "^4.9.0",
    "react-query": "^3.39.3",
    "react-quill": "^2.0.0",
    "styled-components": "^5.3.11",
    "styled-normalize": "^8.0.7",
    "typescript": "5.0.4"
  },
  "devDependencies": {
    "@storybook/addon-essentials": "^7.0.18",
    "@storybook/addon-interactions": "^7.0.18",
    "@storybook/addon-links": "^7.0.18",
    "@storybook/addon-styling": "^1.0.8",
    "@storybook/blocks": "^7.0.18",
    "@storybook/nextjs": "^7.0.18",
    "@storybook/react": "^7.0.18",
    "@storybook/testing-library": "^0.0.14-next.2",
    "@types/bcryptjs": "^2.4.2",
    "@types/isomorphic-fetch": "^0.0.36",
    "@types/styled-components": "^5.1.26",
    "eslint-config-prettier": "^8.8.0",
    "eslint-plugin-storybook": "^0.6.12",
    "husky": "^8.0.3",
    "lint-staged": "^13.2.2",
    "storybook": "^7.0.18"
  }
}`

**Example of how I used nextauth features in my code:**
`
import { useSession } from "next-auth/react";
export function CreatePost() {
  const { status } = useSession();
 
  useEffect(() => {
    if (status === "unauthenticated") Router.replace(Routes.home);
  });

  if (status === "loading") return <div>Loading...</div>;
  if (status === "authenticated")
    return //...`
    
    
 **In .env.local file:**  
I have the NEXTAUTH_URL which is set to http://localhost:3000 and NEXTAUTH_SECRET .

**In the Google and Github providers,** 
I set the URL also to http://localhost:3000 .

**In my _app.tsx:**

I have wrapped around the SessionProvider :

import { SessionProvider } from "next-auth/react";

`<SessionProvider session={session}>
   <Component {...pageProps} />
</SessionProvider>`

Everything is working fine when I npm run dev and npm run start. But when I pushed the commit to the PR. That's the error I'm getting.

Can you help me pinpoint what I was doing wrong? Thank you!!

### How to reproduce

//...

### Expected behavior

//...

**Comment by drcdev** [comment]
Any luck with this issue? I have a pretty barebones project with the same issue. I have one component (a login button) and the api route and that's it. Works in Codespaces, but Actions fails with this error. I've set NEXTAUTH_SECRET and NEXTAUTH_URL as secrets, but can't think of much else to try!

**Comment by natanlf** [comment]
Hi. I have the same problem here. If I run 'npm run build' in my machine I don't have problem. I did a integration before a I did the pipeline and works well bu when I try do the deploy using pipeline this error occurred.

I tried one old version, same project withou nextauth and deploy works fine.

I don't know how to solve.

**Reply by gstrobl** [reply]
Did you found a solution?

**Reply by natanlf** [reply]
Hi. I found the solution. This solves: https://next-auth.js.org/configuration/options
